Auth token (access & refresh token)

ChrisChae00 opened the pull request.

ChrisChae00 committed:
Refactor: Replace hardcoded team name with dynamic team name in board service functions

ChrisChae00 committed:
Feature: Add update user teams functionality and corresponding route

ChrisChae00 committed:
Feature: Integrate teams module and update routes for team management

YoungSuChae committed:
Feature: Enhance team management with new routes and functionalities

- Added new routes for creating, updating, retrieving, and deleting teams.
- Implemented member management features including adding and removing members from teams.
- Updated the `Cargo.toml` to include `serde` feature for `chrono`.
- Refactored existing code to accommodate the new team structure and functionalities.

YoungSuChae committed:
Refactor: Update variable naming for team collection in user teams function. Removed unnecessary logging statements.

YoungSuChae committed:
Remove created_at, updated_at, is_active fields from Team model

Simplifies team management and fixes date serialization issues

YoungSuChae committed:
Feature: Implement refresh token functionality and enhance authentication process

- Added support for refresh tokens in the authentication flow.
- Updated `signup` and `login` functions to return access and refresh tokens.
- Introduced a new route for refreshing access tokens.
- Enhanced error handling and response formatting across authentication services.
- Updated dependencies in `Cargo.toml` and `Cargo.lock` to include `sha2` for token hashing.

YoungSuChae committed:
Fixed session limit bug: properly sort and remove oldest refresh token to enforce max 3 sessions per user.

ChrisChae00 committed:
Feature: Implement performance index creation for user and refresh token collections

- Added a new function to create performance indexes in the MongoDB collections for users and refresh tokens.
- Updated the main application to call this function during startup, ensuring indexes are created for improved query performance.
- Adjusted the refresh token expiration to 30 days and refactored session management logic to handle active sessions more efficiently.

ChrisChae00 committed:
Merge branch 'dev' into auth-token

tonypierre commented on src/main.rs:
```diff
@@ -13,6 +14,12 @@ mod utils;
 async fn main() -> std::io::Result<()> {
     let state = AppState::build().await;
+    
+    match create_performance_indexes(&state.db).await {
```
????

tonypierre reviewed (changes requested):
You need to clean this PR.

As far as I see this PR does 3 things

1. Refresh and Auth Token
2. Create Indexes
3. Fix Custom Errors

ChrisChae00 committed:
Refactor/database index management (#41)

* refactor: separate index creation into dedicated module

- Move index creation logic from auth service to db/indexes module
- Add duplicate index prevention
- Remove unnecessary logging

* refactor: streamline index creation in performance indexes function

* refactor: move index creation to AppState initialization

Move database index creation from main.rs to AppState::build() for better separation of concerns and centralized database initialization.

* feat: implement database index creation with macros

Add create_index! macro and create_performance_indexes function
for automated MongoDB index management during server startup

ChrisChae00 committed:
format

ChrisChae00 committed:
formatting
